contacts/views.py

```diff
@@ -175,12 +177,38 @@ def signup(request):
     if request.method == 'POST':
         form = SignUpForm(request.POST)
         if form.is_valid():
-            form.save()
-            username = form.cleaned_data.get('username')
-            raw_password = form.cleaned_data.get('password1')
-            user = authenticate(username=username, password=raw_password)
-            login(request, user)
-            return HttpResponseRedirect('/user_index/')
+            user = form.save(commit=False)
+            user.is_active = False
+            user.save()
+            current_site = get_current_site(request)
+            mail_subject = 'Activate your account.'
+            account_activation_token = TokenGenerator()
+            message = render_to_string('contacts/acc_active_email.html', {
+                'user': user,
+                'domain': current_site.domain,
+                'uid': urlsafe_base64_encode(force_bytes(user.pk)).decode(),
+                'token': account_activation_token.make_token(user),
+            })
+            to_email = request.POST.get('email')
+            send.delay(mail_subject, to_email, message)
+            return HttpResponse('Please confirm your email address to complete the registration')
     else:
         form = SignUpForm()
     return render(request, 'contacts/signup.html', {'form': form})
+
+
+def activate(request, uidb64, token):
+    account_activation_token = TokenGenerator()
+    try:
+        uid = force_text(urlsafe_base64_decode(uidb64))
+        user = User.objects.get(pk=uid)
+    except(TypeError, ValueError, OverflowError, User.DoesNotExist):
```

Comment:
Why are you catching 3 types of errors?

Reply:
i am catching three different error because:
may user used already activate email url
may user enter manually a token
may our user enter token which limits greater than uidb64 
that;s why i am catching three erros.